pkgs/by-name/do/dosbox-staging/package.nix

```diff
@@ -1,144 +1,87 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-, fetchpatch
-, SDL2
-, SDL2_image
-, SDL2_net
-, alsa-lib
-, copyDesktopItems
-, darwin
-, fluidsynth
-, glib
-, gtest
-, iir1
-, libGL
-, libGLU
-, libjack2
-, libmt32emu
-, libogg
-, libpng
-, libpulseaudio
-, libslirp
-, libsndfile
-, makeDesktopItem
-, makeWrapper
-, meson
-, ninja
-, opusfile
-, pkg-config
-, speexdsp
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  SDL2,
+  SDL2_image,
+  SDL2_net,
+  alsa-lib,
+  darwin,
+  fluidsynth,
+  glib,
+  gtest,
+  iir1,
+  libGL,
+  libGLU,
+  libjack2,
+  libmt32emu,
+  libogg,
+  libpng,
+  libpulseaudio,
+  libslirp,
+  libsndfile,
+  makeWrapper,
+  meson,
+  ninja,
+  opusfile,
+  pkg-config,
+  speexdsp,
+  nix-update-script,
```

Comment:
```suggestion
  gitUpdater,
```

Reply:
I tested `nix-update-script` and it seems to do its job fine, even the nixpkgs manual recommends nix update script. Is there a benefit of switching over to `gitUpdater`?

Reply:
0. I would not say the manual recommends it, it just says it is common.
1. `gitUpdater` is in-house, while nix-update-script is (a wrapper for) an external tool, https://github.com/Mic92/nix-update
2. I am not sure if we should rely on a tool that explicitly say it is "better suited for interactive usage that might require user-intervention".

Reply:
I don't think we shouldn't use a tool just because it's not "in-house". nix-update-script is smart enough to detect things like `rev-prefix`, on its own and more flexible. 

We already use tools like `nixpkgs-review` in nixpkgs to review PRs. It shouldn't matter whether it's external. The developer @Mic92 is well known contributor to Nixpkgs.

The disclaimer indeed has that warning, however nix-update-script is used by a lot of packages and so far I have not noticed any breakage. Every PR on Nixpkgs is reviewed by a maintainer anyway, and merged by a well trusted commiter (except for nixpkgs-merge-bot, we have that now). It's not like someone's opening a Pandora's box. 

I would like a third opinion on this, if possible.

Reply:
Do you want to add yourself to maintainers?

Reply:
Do you want to add yourself to `meta.maintainers`?

Reply:
Nope.

Reply:
> I don't think we shouldn't use a tool just because it's not "in-house".

I am not prohibiting anyone of using it. Indeed it can be used as a standalone tool.

I am just not wanting to use it without understanding its dark magic in an expression I will keep maintenance.

> nix-update-script is smart enough to detect things like rev-prefix, on its own and more flexible.

Well, not to being offensive, but if the upstream developer does not treat the tool as suitable for batch usage, I would not use it as a batch tool in production. 
Who will take responsibility in case of failure?

That being said, nix-update itself respects the update script already defined by the package. It looks  like an extra point to me.

> It shouldn't matter whether it's external.

The internal tool is already present in the Nixpkgs tree, not requiring an extra download trip and reducing the dependency counting.
Well, this is not a strong argument since no update script can run offline - but can be useful if running a batch updater.

> Every PR on Nixpkgs is reviewed by a maintainer anyway, and merged by a well trusted commiter

Famous last words...

Reply:
I was hoping if we could reach an agreement, but well, it is what it is.
If upstream versioning scheme changes in the foreseeable future, nix-update-script would still work, because that's how flexible it is.

I am not planning to mass update packages using the tool, nor I have seen nixpkgs mass updates packages (unless in very specific cases such as GNOME 46/issues).

Even if ryantm-r ends up auto bumping this, along with other packages in batch, I am positive that the PR will atleast be reviewed by a human.

Reply:
The agreement is already done. I will defer to your judgement and depart from all responsibility.

There are way more problems to be solved before losing energy and sleep time on what is the best dark magic updater.